lib/sinatra/soap/request.rb

```diff
@@ -27,14 +40,21 @@ def action
         orig_params[:action] = env['HTTP_SOAPACTION'].to_s.gsub(/^"(.*)"$/, '\1').to_sym
       end
 
-
-      def params 
+      def params
         return orig_params[:soap] unless orig_params[:soap].nil?
         rack_input = env["rack.input"].read
         env["rack.input"].rewind
         orig_params[:soap] = nori.parse(rack_input)[:Envelope][:Body][action]
       end
 
+      def header
+        return orig_params[:soap_header] unless orig_params[:soap_header].nil?
+        rack_input = env["rack.input"].read
+        env["rack.input"].rewind
```

Comment:
never understood the point of this rule, afaik now days compiler is smart enough to determine if have interpolations or not...
 Will update it when code is production ready